app/models/user_mentee_application_cohort.rb

```diff
@@ -0,0 +1,29 @@
+# == Schema Information
+#
+# Table name: user_mentee_application_cohorts
+#
+#  id                :bigint           not null, primary key
+#  active            :boolean          default(TRUE), not null
+#  active_date_range :daterange        not null
+#  created_at        :datetime         not null
+#  updated_at        :datetime         not null
+#
+class UserMenteeApplicationCohort < ApplicationRecord
+  has_many :user_mentee_applications, dependent: nil
+
+  validates :active_date_range, presence: true
+
+  after_save :deactivate_other_cohorts!, if: :active?
+
+  def self.active
+    find_by(active: true)
+  end
+
+  private
+
+  def deactivate_other_cohorts!
+    UserMenteeApplicationCohort.excluding(self).find_each do |cohort|
```

Comment:
can we just do something like `UserMenteeApplicationCohort.excluding(self).update_all(active: false)`?

Reply:
We could. Rubocop warns for use of `update_all` because validations don't run. I could of course just tell it to ignore this method. Don't think that would ever cause a problem in this case.

Reply:
I wonder - if we did it in a single transaction, does that give the benefit we're looking for while also maintaining validations?

Reply:
certain cases I am fine with using update_all because it is just faster. That generally goes when enabling/disabling something or mass updating. This is one of the cases where it is fine.